processing/src/main/java/org/apache/carbondata/processing/store/TablePage.java

```diff
@@ -65,20 +68,20 @@
     this.pageSize = pageSize;
     keyColumnPage = new KeyColumnPage(pageSize,
         model.getSegmentProperties().getDimensionPartitions().length);
-    noDictDimensionPage = new VarLengthColumnPage[model.getNoDictionaryCount()];
+    noDictDimensionPage = new PrimitiveColumnPage[model.getNoDictionaryCount()];
     for (int i = 0; i < noDictDimensionPage.length; i++) {
-      noDictDimensionPage[i] = new VarLengthColumnPage(pageSize);
+      noDictDimensionPage[i] = new PrimitiveColumnPage(DataType.STRING, pageSize);
```

Comment:
Primitive column page should only contains primitive data types, it should not contain string or decimal. So better create another page for string and decimal.